DependencyInjection/Security/Factory/OAuthAuthenticatorFactory.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the HWIOAuthBundle package.
+ *
+ * (c) Hardware Info <opensource@hardware.info>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HWI\Bundle\OAuthBundle\DependencyInjection\Security\Factory;
+
+use HWI\Bundle\OAuthBundle\Security\Http\Authenticator\OAuthAuthenticator;
+use Symfony\Bundle\SecurityBundle\DependencyInjection\Security\Factory\AuthenticatorFactoryInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Vadim Borodavko <vadim.borodavko@gmail.com>
+ */
+class OAuthAuthenticatorFactory extends OAuthFactory implements AuthenticatorFactoryInterface
```

Comment:
Can it be `final` ?

Reply:
Yes, it's not supposed to be inherited, because this class is hard-coded in `HWIOAuthBundle::build()` method.